src/Strategy/Ruby/Errors.hs

```diff
@@ -0,0 +1,45 @@
+module Strategy.Ruby.Errors (
+  BundlerMissingLockFile (..),
+  RubyMissingEdges (..),
+  RubyMissingDepClassification (..),
+
+  -- * Reference docs
+  bundlerLockFileRationaleUrl,
+  rubyFossaDocUrl,
+) where
+
+import App.Docs (strategyLangDocUrl)
+import Data.Text (Text)
+import Diag.Diagnostic (ToDiagnostic, renderDiagnostic)
+import Path
+import Prettyprinter (Pretty (pretty), indent, viaShow, vsep)
+
+bundlerLockFileRationaleUrl :: Text
+bundlerLockFileRationaleUrl = "https://bundler.io/rationale.html#sharing-your-application-with-other-developers"
+
+rubyFossaDocUrl :: Text
+rubyFossaDocUrl = strategyLangDocUrl "ruby/ruby.md"
+
+newtype BundlerMissingLockFile = BundlerMissingLockFile (Path Abs File)
+instance ToDiagnostic BundlerMissingLockFile where
+  renderDiagnostic (BundlerMissingLockFile path) =
+    vsep
+      [ "We could not find or parse Gemfile.lock corresponding to Gemfile: " <> viaShow path
+      , ""
+      , "Ensure your ruby project is built, and has lockfile prior to running fossa."
+      , "If you are using bundler, you can perform: `bundler install` to generate Gemfile.lock."
+      , ""
+      , "Refer to:"
+      , indent 2 $ pretty $ "- " <> bundlerLockFileRationaleUrl
+      , indent 2 $ pretty $ "- " <> rubyFossaDocUrl
+      ]
+
+data RubyMissingEdges = RubyMissingEdges
+instance ToDiagnostic RubyMissingEdges where
+  renderDiagnostic (RubyMissingEdges) =
+    "Could not infer edges between dependencies."
```

Comment:
These errors could use some context.  As a user, I would ask "**Why** can't edges be inferred?"  Same question for the error below: "**Why** can't deps be classified as direct/deep?"

Reply:
It's not clear that the missing `Gemfile.lock` is what causes this.

Reply:
It's not necessarily missing Gemfile.lock, which may cause this. It could be a parser on Gemfile.lock, some ReadFS  error (no permission etc).

Reply:
Maybe - `Could not infer edges between dependencies because Gemfile.lock analysis failed.` can suffice?

Reply:
I think your suggestion is better than the current text, but I think it would be worth making this wordier to explain those possible causes.

"Could not infer edges between dependencies because Gemfile.lock analysis failed. Two possible reasons this can happen are that Gemfile.lock does not exist or we do not have read permission for Gemfile.lock."